molbe/solver.py

```diff
@@ -139,7 +145,15 @@ def be_func(pot, Fobjs, Nocc, solver, enuc, hf_veff=None,
             
         elif solver=='SHCI':
             from pyscf.shciscf import shci
-        
+
+            if scratch is None and be_var.CREATE_SCRATCH_DIR:
+                tmp = os.path.join(be_var.SCRATCH, str(os.getpid()), str(fobj.dname))
+            elif scratch is None:
+                tmp = be_var.SCRATCH
+            else:
+                tmp = os.path.join(scratch, str(os.getpid()), str(fobj.dname))
+            if not os.path.isdir(tmp):
+                os.system('mkdir -p '+tmp)
```

Comment:
`tmp` isn't referenced afterwards, is it?

Reply:
`tmp` is used by the DMRG solver in place of scratch to allow for easy removal afterwards if `force_cleanup=True`. This is a ham-fisted way of doing cleanup, but it ensures no unintended files are removed in the process.